### ISSUE TITLE
[Enhancement] Support self-supervised pre-training

### ISSUE THREAD

**Issue by athewsey** [aws-samples/amazon-textract-transformer-pipeline#3]
Notebook 2 currently has a placeholder section discussing how model accuracy might be improved by self-supervised pre-training (such as [masked language modelling](https://huggingface.co/transformers/task_summary.html#masked-language-modeling)) on a broader corpus of unlabelled data, before fine-tuning on the end task with annotations.

However, the model training scripts in `notebooks/src` aren't set up for that yet.

It'd be great if we can flesh this out to a point where users can optionally run a pre-training job with Textract JSON only (ideally still with the ability to start from a pre-trained model from HF model zoo); and then use that training job instead of the model zoo models, as the starting point for the fine-tuning task.
